linux_efilib.c

```diff
@@ -10,11 +10,33 @@
 #include <wchar.h>
 
 UINTN Print(IN CONST wchar_t *fmt, ...) {
+	// efilib's 'Print' has '%s' or '%ls' for wide and '%a' for ascii.
+	// libc needs '%ls' for wide and '%s' for ascii.
+	// callers must use '%a' and '%ls'; we replace '%a' with '%s'.
+	if (wcsstr(fmt, L"%s") != NULL) {
+		wprintf(L"ERROR: cannot use '%%s' in fmt string: %ls\n", fmt);
+		exit(1);
+	}
+
+	size_t fmtlen = wcslen(fmt);
+	wchar_t *fixedfmt = malloc((fmtlen*sizeof(wchar_t))+sizeof(wchar_t));
```

Comment:
I know this is just testing code, but can you check the return value of malloc here?

Reply:
ack

Reply:
done.